src/frontend/apps/markdown/LanguageSelector/index.spec.tsx

```diff
@@ -0,0 +1,94 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+
+import { wrapInIntlProvider } from 'utils/tests/intl';
+
+import LanguageSelector from '.';
+
+describe('<LanguageSelector />', () => {
+  it('displays languages', async () => {
+    window.scrollTo = jest.fn(); // required to test, see grommet
+
+    const onLanguageChange = jest.fn();
+
+    const { rerender } = render(
+      wrapInIntlProvider(
+        <LanguageSelector
+          currentLanguage={'en'}
+          onLanguageChange={onLanguageChange}
+          disabled={false}
+        />,
+      ),
+    );
+
+    userEvent.click(screen.getByRole('button', { name: 'Language: en' }));
+    userEvent.click(screen.getByTestId('drop_language_fr'));
```

Comment:
```suggestion
    userEvent.click(screen.getByText('fr'));
```

I suggest you to use what the user is seeing. Here it will click on the `fr` text, you can use it in your test. And you can made this change in all this file. And then remove the `data-testid` in your component.